gateway-protocol/src/main/proto/gateway.proto

```diff
@@ -155,7 +155,37 @@ message CreateProcessInstanceWithResultResponse {
   // JSON document
   // consists of visible variables in the root scope
   string variables = 5;
+}
 
+message EvaluateDecisionRequest {
+  // the unique key identifying the decision (e.g. returned from a decision
+  // in the DeployResourceResponse message)
+  int64 decisionKey = 1;
+  // the decision ID
+  string decisionId = 2;
+  // JSON document that will instantiate the variables for the decision; it
+  // must be a JSON object, as variables will be mapped in a key-value fashion.
+  // e.g. { "a": 1, "b": 2 } will create two variables, named "a" and
+  // "b" respectively, with their associated values. [{ "a": 1, "b": 2 }] would
+  // not be a valid argument, as the root of the JSON document is an array and
+  // not an object.
+  string variables = 3;
+}
+
+message EvaluateDecisionResponse {
+  // the unique key identifying the decision (e.g. returned from a decision
+  // in the DeployResourceResponse message)
+  int64 decisionKey = 1;
+  // the decision ID
+  string decisionId = 2;
+  // the version of the decision which was evaluated
+  int32 version = 3;
+  // JSON document that will instantiate the result of the decision; it
+  // will be a JSON object, as the result output will be mapped in a key-value
+  // fashion, e.g. { "a": 1 }.
+  string result = 4;
+  // an optional message describing why the decision evaluation failed
+  string failureMessage = 5;
```

Comment:
❌ We should also add the `failedDecisionId` to indicate which decision failed.

For example, when we evaluate decision A, which requires decision B, and the evaluation of decision A fails at decision B, we should clarify this to the user. This will help the user pinpoint what went wrong and how to resolve the issue.

Reply:
:thinking: This one will be confusing if the `intermediateResults` aren't implemented, as we will provide a `failedDecisionId` but the user won't have access to the "audit trail" to understand where this failed decision ID is coming from.

Reply:
I'm fine with keeping it split. It might keep the reviews a bit more concise than large pull requests with everything at once. Feel free to add to the failedDecisionId in the upcoming pull request.